tests/Fixer/ControlStructure/EmptyLoopBodyFixerTest.php

```diff
@@ -0,0 +1,144 @@
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Fixer\ControlStructure;
+
+use PhpCsFixer\Fixer\ControlStructure\EmptyLoopBodyFixer;
+use PhpCsFixer\Tests\Test\AbstractFixerTestCase;
+
+/**
+ * @author SpacePossum
+ *
+ * @internal
+ *
+ * @covers \PhpCsFixer\Fixer\ControlStructure\EmptyLoopBodyFixer
+ */
+final class EmptyLoopBodyFixerTest extends AbstractFixerTestCase
+{
+    /**
+     * @dataProvider provideFixCases
+     */
+    public function testFixConfig(string $expected, ?string $input = null, ?array $config = null): void
+    {
+        if (null === $config) {
+            $this->doTest($expected, $input);
+
+            $this->fixer->configure(['style' => EmptyLoopBodyFixer::STYLE_SEMICOLON]);
+
+            if (null === $input) {
+                $this->doTest($expected, $input);
+            } else {
+                $this->doTest($input, $expected);
+            }
+        } else {
+            $this->fixer->configure($config);
+            $this->doTest($expected, $input);
+        }
+    }
+
+    public function provideFixCases()
+    {
+        yield 'simple "while"' => [
+            '<?php while(foo()){}',
+            '<?php while(foo());',
+        ];
+
+        yield 'simple "for"' => [
+            '<?php for($i = 0;foo();++$i){}',
+            '<?php for($i = 0;foo();++$i);',
+        ];
+
+        yield 'simple "foreach"' => [
+            '<?php foreach (Foo() as $f){}',
+            '<?php foreach (Foo() as $f);',
+        ];
+
+        yield '"while" followed by "do-while"' => [
+            '<?php while(foo(static function(){})){} do{ echo 1; }while(bar());',
+            '<?php while(foo(static function(){})); do{ echo 1; }while(bar());',
+        ];
+
+        yield 'empty "while" after "if"' => [
+            '<?php
+if ($foo) {
+    echo $bar;
+} while(foo()){}
+',
+            '<?php
+if ($foo) {
+    echo $bar;
+} while(foo());
+',
+        ];
+
+        yield 'nested and mixed loops' => [
+            '<?php
+
+do {
+    while($foo()) {
+        while(B()){} // fix
+        for($i = 0;foo();++$i){} // fix
+
+        for($i = 0;foo();++$i) {
+            foreach (Foo() as $f){} // fix
+        }
+    }
+} while(foo());
+',
+            '<?php
+
+do {
+    while($foo()) {
+        while(B()); // fix
+        for($i = 0;foo();++$i); // fix
+
+        for($i = 0;foo();++$i) {
+            foreach (Foo() as $f); // fix
+        }
+    }
+} while(foo());
+',
+        ];
+
+        yield 'not empty "while"' => [
+            '<?php while(foo()){ bar(); };',
+        ];
+
+        yield 'not empty "for"' => [
+            '<?php for($i = 0; foo(); ++$i){ bar(); }',
+        ];
+
+        yield 'not empty "foreach"' => [
+            '<?php foreach (foo() as $f){ echo 1; }',
+        ];
+
+        yield [
+            '<?php while (foo1())
+;
+
+
+
+echo 1;
+',
+            '<?php while (foo1())
+{
+
+}
+
+echo 1;
+',
+            ['style' => 'semicolon'],
```

Comment:
Could you add tests with comments like e.g.
```php
while (foo()) {
    // noop
}
```
In this example, the body shouldn't be considered empty IMO.

Reply:
why not, if you use braces as style it won't be touched, if you want to use `;` than it will be fixed and leave the comment,
if you want to use `;` but have exceptions as this one than don't use the rule

Reply:
Fair enough.